cmd/kueuectl/app/list/list_clusterqueue.go

```diff
@@ -0,0 +1,252 @@
+	// Active means the cluster queue has kueue.ClusterQueueActive condition with status=metav1.ConditionTrue
+	Active string
+
+	Client kueuev1beta1.KueueV1beta1Interface
+
+	PrintObj printers.ResourcePrinterFunc
+
+	// ctx is the context for the command
+	ctx context.Context
+
+	genericiooptions.IOStreams
+}
+
+func NewClusterQueueOptions(streams genericiooptions.IOStreams) *ClusterQueueOptions {
+	return &ClusterQueueOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewClusterQueueCmd(clientGetter genericclioptions.RESTClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewClusterQueueOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "clusterqueue [--selector key1=value1] [--field-selector key1=value1] [--active=*|true|false]",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"cq"},
+		Short:                 cqShort,
+		Long:                  cqLong,
+		Example:               cqExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter, cmd, args))
+			cobra.CheckErr(o.Validate())
+			cobra.CheckErr(o.Run())
+		},
+		SuggestFor: []string{"ps"},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addClusterQueueActiveFilterFlagVar(cmd, &o.Active)
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *ClusterQueueOptions) Complete(clientGetter genericclioptions.RESTClientGetter, cmd *cobra.Command, args []string) error {
+	var err error
+	o.ctx = cmd.Context()
+
+	config, err := clientGetter.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	clientset, err := versioned.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	o.Client = clientset.KueueV1beta1()
+
+	if !o.PrintFlags.OutputFlagSpecified() {
+		o.PrintObj = printClusterQueueTable
+	} else {
+		printer, err := o.PrintFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+		o.PrintObj = printer.PrintObj
+	}
+
+	if len(args) > 0 {
+		activeFlag, err := cmd.Flags().GetString("active")
+		if err != nil {
+			return err
+		}
+		o.Active = activeFlag
+	}
+
+	return nil
+}
+
+func (o *ClusterQueueOptions) Validate() error {
+	if !o.validActiveFlagOptionProvided() {
+		return fmt.Errorf("invalid value for --active flag: %s", o.Active)
+	}
+	return nil
+}
+
+func (o *ClusterQueueOptions) validActiveFlagOptionProvided() bool {
+	return o.Active == string(clusterQueueActiveAll) ||
+		o.Active == string(clusterQueueActiveEmpty) ||
+		o.Active == string(clusterQueueActiveTrue) ||
+		o.Active == string(clusterQueueActiveFalse)
+}
+
+// Run prints the cluster queues.
+func (o *ClusterQueueOptions) Run() error {
+	opts := metav1.ListOptions{LabelSelector: o.LabelSelector, FieldSelector: o.FieldSelector}
+	list, err := o.Client.ClusterQueues().List(o.ctx, opts)
```

Comment:
Probably it's not the case for cluster queues, but we should maybe take into account paging of the lists.